controller/src/main/java/io/pravega/controller/store/stream/StoreException.java

```diff
@@ -114,12 +134,42 @@ public NodeNotEmptyException() {
     }
 
     /**
+     * Exception type when you are attempting to update a stale value.
+     */
+    public static class WriteConflictException extends StoreException {
```

Comment:
An exception which is retryable in one context may not be retryable in another context, hence decorating exceptions with retryable trait may not be always possible. We could instead have pre-defined predicates that classify retryable exceptions. These predicates can be reused in multiple contexts.